code/game/objects/structures/sensor_tower.dm

```diff
@@ -61,3 +61,101 @@
 			SSminimaps.remove_marker(src)
 			SSminimaps.add_marker(src, z, MINIMAP_FLAG_ALL, "neutral_zone")
 
+/obj/structure/sensor_tower_patrol
+	name = "sensor tower"
+	desc = "A tall tower with a sensor array at the top and a control box at the bottom. Has a lengthy activation process."
+	icon = 'icons/obj/structures/sensor.dmi'
+	icon_state = "sensor"
+	obj_flags = NONE
+	density = TRUE
+	layer = BELOW_OBJ_LAYER
+	resistance_flags = RESIST_ALL
+	var/current_timer
+	var/generate_time = 150 SECONDS
+	var/activate_time = 5 SECONDS // time to start the activation
+	var/deactivate_time = 10 SECONDS // time to stop the activation proccess
+	var/static/id = 1
+	var/towerid
+	var/activated = FALSE // if all segments are finished
+
+/obj/structure/sensor_tower_patrol/Initialize()
+	. = ..()
+	name += " " + num2text(id)
+	towerid = id
+	id++
+	update_icon()
+
+/obj/structure/sensor_tower_patrol/update_icon_state()
+	icon_state = initial(icon_state)
+	if(current_timer || activated)
+		icon_state += "_loyalist"
+
+/obj/structure/sensor_tower_patrol/attack_hand(mob/living/user)
+	if(!ishuman(user))
+		return
+	if(user.do_actions)
+		user.balloon_alert(user, "You are already doing something!")
+		return
+	if(user.faction != FACTION_TERRAGOV)
+		if(current_timer)
+			balloon_alert(user, "You begin to stop the activation process!")
+			if(!do_after(user, deactivate_time, TRUE, src))
+				return
+			balloon_alert(user, "You stop the activation process!")
+			deactivate()
+		else if(activated)
+			balloon_alert(user, "This sensor tower is already fully activated, you cannot deactivate it!")
+		else
+			balloon_alert(user, "This sensor tower is not activated yet, don't let it be activated!")
+		return
+	if(activated)
+		balloon_alert(user, "[src] is already fully activated!")
+		return
+	if(current_timer)
+		balloon_alert(user, "[src] is currently activating!")
+		return
+	balloon_alert_to_viewers("[user] starts to activate [src]!")
+	if(!do_after(user, activate_time, TRUE, src))
+		return
+	balloon_alert_to_viewers("[user] activates [src]!")
```

Comment:
dont use [src] in baloon alerts and even generally avoid inserting names
these should be as short as possible, and remember that players can see where it is on the map (so no specified what is being affected needed)